lumicks/pylake/kymotracker/kymoline.py

```diff
@@ -1,6 +1,67 @@
 import numpy as np
 
 
+def export_kymolines_to_csv(filename, kymo_lines, dt, dx, delimiter, sampling_width):
+    """Export kymolines to a csv file.
+
+    Parameters
+    ----------
+    filename : str
+        Filename to output kymolines to.
+    kymo_lines : KymoLines[pylake.KymoLine]
+        List of kymograph traces to export.
+    dt : float
+        Calibration for the time axis.
+    dx : float
+        Calibration for the coordinate axis.
+    delimiter : str
+        Which delimiter to use in the csv file.
+    sampling_width : int or None
+        When supplied, this will sample the source image around the kymograph line and export the summed intensity with
+        the image. The value indicates the number of pixels in either direction to sum over.
+    """
+    if not kymo_lines:
+        raise RuntimeError("No kymolines to export")
+
+    idx = np.hstack([idx * np.ones(len(line.time_idx)) for idx, line in enumerate(kymo_lines)])
+    coords_idx = np.hstack([line.coordinate_idx for line in kymo_lines])
+    times_idx = np.hstack([line.time_idx for line in kymo_lines])
+
+    data, header, fmt = [], [], []
+
+    def store_column(column_title, format_string, new_data):
+        data.append(new_data)
+        header.append(column_title)
+        fmt.append(format_string)
+
+    store_column("line index", "%d", idx)
+    store_column("time (pixels)", "%.18e", times_idx)
+    store_column("coordinate (pixels)", "%.18e", coords_idx)
+
+    if dt:
+        store_column("time", "%.18e", [times_idx * dt])
+
+    if dx:
+        store_column("coordinate", "%.18e", [coords_idx * dx])
+
+    if sampling_width is not None:
+        store_column(f"counts (summed over {2 * sampling_width + 1} pixels)", "%d",
+                     np.hstack([line.sample_from_image(sampling_width) for line in kymo_lines]))
+
+    data = np.vstack(data).T
+    np.savetxt(filename, data, fmt=fmt, header=delimiter.join(header), delimiter=delimiter)
+
+
+def import_kymolines_from_csv(filename, image, delimiter=';'):
+    data = np.loadtxt(filename, delimiter=delimiter)
+    assert len(data.shape) == 2, "Invalid file format"
+    assert data.shape[0] > 2, "Invalid file format"
+
+    indices = data[:, 0]
+    lines = np.unique(indices)
+    return KymoLines(KymoLine(data[indices == k, 1], data[indices == k, 2], image) for k in lines)
```

Comment:
Maybe add a comment or docstring indicating the labels for each column of the data. I find it helpful rather than having to refer to back to the saving function.

Reply:
Yeah, I'll add it. Only risk is going out of date with the implementation, but we'll just have to be vigilant of this. Added the following docstring:

```python
    """Import kymolines from csv

    Parameters
    ----------
    filename : str
        filename to import from
    image : array_like
        2D image that these lines were tracked from
    delimiter : str
        A delimiter that delimits the column data.

    The file format contains a series of columns as follows:
    line index, time (pixels), coordinate (pixels), time (optional), coordinate (optional), sampled_counts (optional)"""
```